functions.opengraph.php

```diff
@@ -100,10 +104,12 @@ function jetpack_og_tags() {
 
 		$tags['og:url']         = get_permalink( $data->ID );
 		if ( ! post_password_required() ) {
-			if ( ! empty( $data->post_excerpt ) ) {
-				$tags['og:description'] = preg_replace( '@https?://[\S]+@', '', strip_shortcodes( wp_kses( $data->post_excerpt, array() ) ) );
+			$excerpt = get_the_excerpt( $data );
+			if ( ! empty( $excerpt ) && ! empty( $data->post_excerpt ) ) {
+				$tags['og:description'] = preg_replace( '@https?://[\S]+@', '', $excerpt
+ );
 			} else {
-				$exploded_content_on_more_tag = explode( '<!--more-->', $data->post_content );
+				$exploded_content_on_more_tag = explode( '<!--more-->', apply_filters( 'the_content', $data->post_content ) );
```

Comment:
understood! thanks for the additional context.

so the tradeoff is, either tolerate the problem here (#6181) until existing code and plugins update and use a new jetpack-specific filter to prevent it, or regress on #2899 and include appended content again.

neither sounds ideal. i'm biased since this is biting me and #2899 isn't...but i'm fine with the new filter approach, and you all know wordpress decision making better than me. i'll take a stab at it. let me know if you'd rather do it yourself!